refactor(napi/oxlint): reverse args of `ExternalLinter::new`

overlookmotel opened the pull request:
Pure refactor. Reverse order of arguments to `ExternalLinter::new`, to match order of fields in `ExternalLinter` struct.

camc314 commented:
### Merge activity

* **Jul 28, 10:47 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(napi/oxlint): reverse args of `ExternalLinter::new` (#12566)

Pure refactor. Reverse order of arguments to `ExternalLinter::new`, to match order of fields in `ExternalLinter` struct.
